src/andershoest_ex/pa02/chutes_simulations.py

```diff
@@ -0,0 +1,257 @@
+
+class Board:
+    # Ladder dictionary and snake dictionary from pa01
+    ladder_dict = {1: 40, 8: 10, 36: 52, 43: 62, 49: 79, 65: 82, 68: 85}
+    snake_dict = {24: 5, 33: 3, 42: 30, 56: 37, 64: 27, 74: 12, 87: 70}
+
+    def __init__(self, ladders=None, snakes=None, goal=90):
+        """
+           Initializes the board.
+
+           Creating a class of the board. Ladders and snakes represented by
+           lists. Position 90 is the endpoint and position 0 is the start
+           position.
+
+           Arguments
+           ---------
+           param start : int 0
+               The first position on the board.
+           param goal : int 90
+               the last position on the board.
+
+           """
+        self.ladders = ladders or Board.ladder_dict
+        self.snakes = snakes or Board.snake_dict
+        self.goal = goal
+
+    def goal_reached(self, position):
+        """
+        Checks if goal is reached
+
+        :param position:
+        :return: bo
+        """
+        return position >= self.goal
+
+    def position_adjustment(self, position):
+        """
+        The method takes as argument the position of the player
+        and returns integer number of positions the player must move
+        forward in case of a ladder or backward in case of a snake
+
+        :param position
+        :return: int
+        """
+        changed_position = 0
+        if position in self.ladders:
+            return self.ladders[position] - position
+        elif position in self.snakes:
+            return self.snakes[position] - position
+        return changed_position
+
+
+class Player:    # AH: correct? or should it be Player(board)? Test OK
+
+    def __init__(self, board):
+        """
+        Initializes the player class.
+
+        :param board
+        """
+        self.board = board
+        self.position = 0
+        self.no_moves = 0    # Added number of moves
+        self.adjustment = 0
+
+    def move(self):
+        if self.board.goal_reached(self.position):
+            return
+        # Stop moving after the goal is reached
+        throw_die = random.randint(1, 6)
+        # number between 1 and 6
+        new_temporary_pos = self.position + throw_die
+        # move player to new position
+        self.adjustment = self.board.position_adjustment(new_temporary_pos)
+        # look for ladders or snakes
+        self.position = new_temporary_pos + self.adjustment
+        # changes position if adjustment is not 0.
+        self.no_moves += 1    # Increasing the number of moves
+
+    def check_goal_reached(self):    # Added method check goal reached
+        if self.board.goal_reached(self.position):
+            return True
+
+
+class ResilientPlayer(Player):
+
+    def __init__(self, board, extra_steps=5):
+        """
+        Initializes ResilientPlayer as Subclass of Player.
+
+        :param: board, extra_steps
+        """
+        super().__init__(board)
+        self.extra_steps = extra_steps
+
```

Comment:
you will need a flag here to know if the player went down a chute, so the next time he should take extra steps